servatrice/src/serversocketinterface.cpp

```diff
@@ -756,18 +756,33 @@ Response::ResponseCode ServerSocketInterface::cmdBanFromServer(const Command_Ban
     if (trustedSources.contains(address,Qt::CaseInsensitive))
         address = "";
 
-    QSqlQuery *query = sqlInterface->prepareQuery("insert into {prefix}_bans (user_name, ip_address, id_admin, time_from, minutes, reason, visible_reason) values(:user_name, :ip_address, :id_admin, NOW(), :minutes, :reason, :visible_reason)");
+    QSqlQuery *query = sqlInterface->prepareQuery("insert into {prefix}_bans (user_name, ip_address, id_admin, time_from, minutes, reason, visible_reason, clientid) values(:user_name, :ip_address, :id_admin, NOW(), :minutes, :reason, :visible_reason, :client_id)");
     query->bindValue(":user_name", userName);
     query->bindValue(":ip_address", address);
     query->bindValue(":id_admin", userInfo->id());
     query->bindValue(":minutes", minutes);
     query->bindValue(":reason", QString::fromStdString(cmd.reason()));
     query->bindValue(":visible_reason", QString::fromStdString(cmd.visible_reason()));
+    query->bindValue(":client_id", QString::fromStdString(cmd.clientid()));
     sqlInterface->execSqlQuery(query);
 
     servatrice->clientsLock.lockForRead();
     QList<ServerSocketInterface *> userList = servatrice->getUsersWithAddressAsList(QHostAddress(address));
     ServerSocketInterface *user = static_cast<ServerSocketInterface *>(server->getUsers().value(userName));
+
+    if (userName.isEmpty() && address.isEmpty()) {
+        QSqlQuery *query = sqlInterface->prepareQuery("select name from {prefix}_users where clientid = :client_id");
+        query->bindValue(":client_id", QString::fromStdString(cmd.clientid()));
+        sqlInterface->execSqlQuery(query);
+        while (query->next()){
+            userName = query->value(0).toString();
+            ServerSocketInterface *user = static_cast<ServerSocketInterface *>(server->getUsers().value(userName));
+            if (user && !userList.contains(user))
```

Comment:
If you convert the QList to a QSet then you don't need to do this - you can just `add` it whether or not it was already there.